app/src/main/java/fr/free/nrw/commons/repository/UploadRemoteDataSource.java

```diff
@@ -0,0 +1,189 @@
+package fr.free.nrw.commons.repository;
+
+import fr.free.nrw.commons.category.CategoriesModel;
+import fr.free.nrw.commons.category.CategoryItem;
+import fr.free.nrw.commons.contributions.Contribution;
+import fr.free.nrw.commons.filepicker.UploadableFile;
+import fr.free.nrw.commons.nearby.Place;
+import fr.free.nrw.commons.upload.SimilarImageInterface;
+import fr.free.nrw.commons.upload.UploadController;
+import fr.free.nrw.commons.upload.UploadModel;
+import fr.free.nrw.commons.upload.UploadModel.UploadItem;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+
+import java.util.Comparator;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class UploadRemoteDataSource {
```

Comment:
Thank you for the method Javadocs. The class itself needs a Javadoc to summarize what it does though, especially if it isn't a regular Fragment/Activity/etc.